index.bs

```diff
@@ -2092,83 +3395,189 @@ partial interface MLGraphBuilder {
+</div>
+
+<details open>
+  <summary>
+    The {{MLGraphBuilder/leakyRelu(options)}} method steps are:
+  </summary>
+  <div algorithm=leakyrelu-options class=algorithm-steps>
+    1. Let |op| be the result of <a>creating an MLActivation</a> given [=this=], `"leakyRelu"` and |options|.
+        1. If that [=exception/throws=] an error, re-[=exception/throw=] the error.
+    1. Return |op|.
+  </div>
+</details>
+
 ### The linear() method ### {#api-mlgraphbuilder-linear}
 Calculate a linear function `y = alpha * x + beta` on the input tensor.
+
 <script type=idl>
 dictionary MLLinearOptions {
   float alpha = 1;
   float beta = 0;
 };
 
 partial interface MLGraphBuilder {
-  MLOperand linear(MLOperand x, optional MLLinearOptions options = {});
+  MLOperand linear(MLOperand input, optional MLLinearOptions options = {});
   MLActivation linear(optional MLLinearOptions options = {});
 };
 </script>
-<div algorithm=linear>
-    **Arguments:**
-        - *x*: an {{MLOperand}}. The input tensor.
-        - *options*: an optional {{MLLinearOptions}}. The optional parameters of the operation.
-            - *alpha*: a {{float}} scalar multiplier, default to 1.
-            - *beta*: a {{float}} scalar addition, default to 0.
 
-    **Returns:**
-        - an {{MLOperand}}. The output tensor of the same shape as *x*.
-        - an {{MLActivation}}. The activation function representing the linear operation.
-
-    <div class="note">
+<div class="note">
+  <details open>
+    <summary>
     The behavior of this operation can be generically emulated from the usage of
     other operations as follow. However, user agents typically have a more
     efficient implementation for it, therefore its usage is encouraged from the
     performance standpoint.
+    </summary>
     <pre highlight="js">
     return builder.add(
               builder.mul(x, builder.constant(options.alpha)),
               builder.constant(options.beta));
     </pre>
-    </div>
+  </details>
 </div>
 
+{{MLLinearOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLLinearOptions>
+    : <dfn>alpha</dfn>
+    ::
+         A {{float}} scalar multiplier.
+         The default value is `1`.
+    : <dfn>beta</dfn>
+    ::
+         A {{float}} scalar addition.
+         The default value is `0`.
+</dl>
+
+#### The {{MLGraphBuilder/linear(input, options)}} method #### {#api-mlgraphbuilder-linear-input-options}
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input tensor.
+        - *options*: an optional {{MLLinearOptions}}. The optional parameters of the operation.
+
+    **Returns:**
+        - an {{MLOperand}}. The output tensor of the same shape as *input*.
+</div>
+
+<details open>
+  <summary>
+    The {{MLGraphBuilder/linear(input, options)}} method steps are:
+  </summary>
+  <div algorithm=linear-input-options class=algorithm-steps>
+    1. If any of the following sub-steps fail, [=exception/throw=] an "{{OperationError}}" {{DOMException}}.
```

Comment:
Consider adding a check for input
```
 1. [=Assert=]: the type of |input| is {{MLOperand}}.
```

Reply:
FWIW, I'd reserve asserts only for non-obvious things, e.g. where a much earlier step in an algorithm has ensured that a number is non-zero, so a division by it is safe.

Asserts like these that repeat the concise Web IDL definition in longer prose are just noise and IMHO not useful for implementers.

Reply:
That's right, it occurred to me, too - on the other side, IMHO it helps readability, especially for "laymen". Often I read a spec and need to click back for types, this saves that. But I do agree, and looks like brevity/conciseness is a top priority nowadays.

I guess removing these asserts will be fine with our impl team, will confirm with them.

Reply:
Moved to #455